docarray/array/storage/base/helper.py

```diff
@@ -0,0 +1,42 @@
+from typing import Iterator
+
+
+class Offset2ID:
+    def __init__(self):
+        self.offset2id = []
+
+    def get_id(self, idx):
+        return self.offset2id[idx]
+
+    def append(self, data):
+        self.offset2id.append(data)
+
+    def update(self, position, data_id):
+        self.offset2id[position] = data_id
+
+    def delete_by_id(self, _id):
+        del self.offset2id[self.offset2id.index(_id)]
+
+    def index(self, _id):
+        return self.offset2id.index(_id)
+
+    def delete_by_offset(self, position):
+        del self.offset2id[position]
+
+    def insert(self, position, data_id):
+        self.offset2id.insert(position, data_id)
```

Comment:
Since the basic data container is a Dict now. I have to decide whether do we need to check the inserted `data_id` is unique?

Reply:
right
It doesn't make sense anyways for storage backends
so we don't allow docs with same ID ?

Reply:
I would prefer to disable documents have same ID